jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -130,13 +141,12 @@
     System.out.println("-----\ndone\ntook " + (end - start) + "ms");
     System.out.println("definitely-derefereced paramters: " + map_str_result.toString());
 
-    writeJarModel(cha, map_mtd_result);
+    writeJarModel(cha, map_mtd_result, aStubXPath);
 
     if (path.endsWith(".jar")) {
-      packJAR(workDir);
+      packJAR(workDir, outPath);
```

Comment:
From an offline discussion with @kageiit, this `.jar`/`.aar` check happens in a lot of places and might be more cleanly handled by a factory pattern.